src/corelibs/U2Core/src/models/GObject.cpp

```diff
@@ -172,18 +172,10 @@ void GObject::setRelationsInDb(QList<GObjectRelation> &list) const {
         const U2DataType refType = U2ObjectTypeUtils::toDataType(relation.ref.objType);
         const bool relatedObjectDbReferenceValid = relation.ref.entityRef.dbiRef.isValid();
 
-        if (U2Type::Unknown == refType || (relatedObjectDbReferenceValid && !(relation.ref.entityRef.dbiRef == entityRef.dbiRef))) {
+        if (U2Type::Unknown == refType || !relatedObjectDbReferenceValid) {
```

Comment:
The removed check checks that self-relations are not allowed. Why do you want to remove it?